Mathlib/CategoryTheory/Enriched/Ordinary.lean

```diff
@@ -0,0 +1,138 @@
+we may require that the type of morphisms `X ⟶ Y` in `C` identify to
+`𝟙_ V ⟶ EnrichedCategory.Hom X Y`. This data shall be packaged in the
+typeclass `EnrichedOrdinaryCategory V C`.
+
+In particular, if `C` is a `V`-enriched category, it is shown that
+the "underlying" category `ForgetEnrichment V C` is equipped with a
+`EnrichedOrdinaryCategory V C` instance.
+
+Simplicial categories are implemented in `AlgebraicTopology.SimplicialCategory.Basic`
+using an abbreviation for `EnrichedOrdinaryCategory SSet C`.
+
+-/
+
+universe v' v u u'
+
+open CategoryTheory Category MonoidalCategory
+
+namespace CategoryTheory
+
+variable (V : Type u') [Category.{v'} V] [MonoidalCategory V]
+  (C : Type u) [Category.{v} C]
+
+/-- An enriched ordinary category is a category `C` that is also enriched
+over a category `V` in such a way that morphisms `X ⟶ Y` in `C` identify
+to morphisms `𝟙_ V ⟶ (X ⟶[V] Y)` in `V`. -/
+class EnrichedOrdinaryCategory extends EnrichedCategory V C where
+  /-- morphisms `X ⟶ Y` in the category identify morphisms
+    `𝟙_ V ⟶ (X ⟶[V] Y)` in `V` -/
+  homEquiv {X Y : C} : (X ⟶ Y) ≃ (𝟙_ V ⟶ (X ⟶[V] Y))
+  homEquiv_id (X : C) : homEquiv (𝟙 X) = eId V X := by aesop_cat
+  homEquiv_comp {X Y Z : C} (f : X ⟶ Y) (g : Y ⟶ Z) :
+    homEquiv (f ≫ g) = (λ_ _).inv ≫ (homEquiv f ⊗ homEquiv g) ≫
+      eComp V X Y Z := by aesop_cat
+
+variable [EnrichedOrdinaryCategory V C] {C}
+
+/-- The bijection `(X ⟶ Y) ≃ (𝟙_ V ⟶ (X ⟶[V] Y))` given by a
+`EnrichedOrdinaryCategory` instance. -/
+def eHomEquiv {X Y : C} : (X ⟶ Y) ≃ (𝟙_ V ⟶ (X ⟶[V] Y)) :=
+  EnrichedOrdinaryCategory.homEquiv
+
+@[simp]
+lemma eHomEquiv_id (X : C) : eHomEquiv V (𝟙 X) = eId V X :=
+  EnrichedOrdinaryCategory.homEquiv_id _
+
+@[reassoc]
+lemma eHomEquiv_comp {X Y Z : C} (f : X ⟶ Y) (g : Y ⟶ Z) :
+    eHomEquiv V (f ≫ g) = (λ_ _).inv ≫ (eHomEquiv V f ⊗ eHomEquiv V g) ≫ eComp V X Y Z :=
+  EnrichedOrdinaryCategory.homEquiv_comp _ _
+
+/-- The morphism `(X' ⟶[V] Y) ⟶ (X ⟶[V] Y)` induced by a morphism `X ⟶ X'`. -/
+noncomputable def eHomWhiskerRight {X X' : C} (f : X ⟶ X') (Y : C) :
+    (X' ⟶[V] Y) ⟶ (X ⟶[V] Y) :=
+  (λ_ _).inv ≫ eHomEquiv V f ▷ _ ≫ eComp V X X' Y
+
+@[simp]
+lemma eHomWhiskerRight_id (X Y : C) : eHomWhiskerRight V (𝟙 X) Y = 𝟙 _ := by
+  simp [eHomWhiskerRight]
+
+@[simp, reassoc]
+lemma eHomWhiskerRight_comp {X X' X'' : C} (f : X ⟶ X') (f' : X' ⟶ X'') (Y : C) :
+    eHomWhiskerRight V (f ≫ f') Y = eHomWhiskerRight V f' Y ≫ eHomWhiskerRight V f Y := by
+  dsimp [eHomWhiskerRight]
+  rw [assoc, assoc, eHomEquiv_comp, comp_whiskerRight_assoc, comp_whiskerRight_assoc, ← e_assoc',
+    tensorHom_def', comp_whiskerRight_assoc, id_whiskerLeft, comp_whiskerRight_assoc,
+    ← comp_whiskerRight_assoc, Iso.inv_hom_id, id_whiskerRight_assoc,
+    comp_whiskerRight_assoc, leftUnitor_inv_whiskerRight_assoc,
+    ← associator_inv_naturality_left_assoc, Iso.inv_hom_id_assoc,
+    ← whisker_exchange_assoc, id_whiskerLeft_assoc, Iso.inv_hom_id_assoc]
```

Comment:
I wonder why the shorter proofs from the old simplicial category file don't work. Is `dsimp` doing more in that file?

Reply:
It was probably due to the fact that most monoidal compatibilities in the category of sets (with chosen products) are just `rfl`.